### DISCUSSION TITLE
Python filter and code block woes

### DISCUSSION THREAD

**Discussion by madsmh** [jgm/pandoc#10476] [Q&A] [answered]
I have this MArkdown file made with Obsidian which contains a code block "```tikz"
I have been trying to adapt an an example of python filters form pandocfilers, and have written this:

`import pandocfilters

def tikzjax(key, value, format, _):
    if key == 'RawBlock':
        [fmt, code] = value
        if fmt == 'tikz':
            return pandocfilters.Para(["Ha!"])

if __name__ == "__main__":
    pandocfilters.toJSONFilter(tikzjax)
`
My intention was to be able to detect tikz codeblocks and manipulate them. The above script was supposed to simply replace the code block with the text "Ha!"However, it does not do anything. Any  help on where I am going wrong would be greatly appreciated.

**Comment by jgm** [comment]
You want to match on CodeBlock, not RawBlock, as this will be parsed as a code block.

**Reply by madsmh** [reply]
Hi, thank you for your reply!
My script now looks like

```
import pandocfilters

def tikzjax(key, value, format, _):
    if key == 'CodeBlock':
        [fmt, code] = value
        if fmt == 'tikz':
            return pandocfilters.Para(["Ha!"])

if __name__ == "__main__":
    pandocfilters.toJSONFilter(tikzjax)

```

but it still does not affect the tikz code blocks. I am using the command:

` pandoc.exe -f markdown -t latex --filter ..\latex_filter.py -s '.\output.md' -o test.tex`

Best, Mads

**Reply by jgm** [reply]
You'll have to do some debugging.  See what value `fmt` has, for one...
I'm sorry, even though I wrote pandocfilters, it was ages ago and I don't remember the details.

**Comment by madsmh** [comment, marked as answer]
HI again,
```
import pandocfilters

def tikzjax(key, value, format, _):
    if key == 'CodeBlock':
        [fmt, code] = value
        if fmt[1} == ['tikz']:
            ## .....

if __name__ == "__main__":
    pandocfilters.toJSONFilter(tikzjax)
```
worked.

Best, Mads
